packages/shared/src/hooks/useJoinSquad.ts

```diff
@@ -41,7 +44,8 @@ export const useJoinSquad = ({
     });
 
     addSquad(result);
-    queryClient.invalidateQueries(['squad', squad.handle]);
+    const queryKey = generateQueryKey(RequestKey.Squad, user, squad.handle);
+    queryClient.invalidateQueries(queryKey);
```

Comment:
Ah found one edge case. @sshanzel 

In `SquadJoinButton` we pass `joinSquad` to `showLogin`, but since in that function reference user is still for anonymous it actually clears the cache for `[squad, anonymous, handle]` and the new key (because user is logged) is `[squad, user.id, handle]`. Which then shows stale page like user is not part of squad. If you refresh the page he is of course. 

We can fix in two ways I think:
- by just invalidating all the squad data for now invalidateQueries([RequestKey.Squad])
- using useRef to always have a fresh instance of `joinSquad` when calling in login success callbacks

Reply:
Don't know if you have any other ideas, we can also go through the case on a quick call.

Reply:
If you now have a new query key by the `anonymous` getting replaced with the user's id, then it should automatically refetch since the query params have changed.

Reply:
Do you have any steps where I can reproduce this?

Reply:
Yeah, but the refetch happens at the moment user logs in (in useSquad) `joinSquad` mutation is done so it gets that user is still not part of squad.